doc/compiler/jitserver/Overview.md

```diff
@@ -22,8 +22,30 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 
 # JITServer Overview
 
+This document provides a high-level overview of JITServer architecture.
+If you are looking for a step-by-step guide on how to use JITServer, read ["Getting started with JITServer"](Usage.md) instead.
+
+As mentioned in the above guide, JITServer introduces two additional personas to JVM: **client** and **server**, with clients outsourcing most of their compilation workload to a server.
+
+## Server (`jitserver` binary)
+
+Server is a JVM where only the JIT compiler is active. It compiles methods for client JVMs.
+
+Server has 64 compilation threads, compared to regular JVM's 7 threads. One server can serve multiple clients at the same time.
```

Comment:
By default the server can use 64 compilation threads. This value can be increased up to 999 with with `-XcompilationThreads<NNN>` command line option.